includes/User.php

```diff
@@ -5007,6 +5007,8 @@ protected function saveOptions() {
 
 		$dbw->upsert('user_properties', $insertRows, [], self::$PROPERTY_UPSERT_SET_BLOCK);
 
+		$this->userPreferences()->save($this->getId());
```

Comment:
We can move it there, just not that there is also `saveOptions()` in `addToDatabase()` so we would need to save settings in both places.